arch/arm/core/cortex_m/mpu/nxp_mpu.c

```diff
@@ -196,27 +193,20 @@ static void nxp_mpu_setup_sram_region(u32_t base, u32_t size)
  */
 void arm_core_mpu_enable(void)
 {
-	if (nxp_mpu_enabled == 0) {
-		/* Enable MPU */
-		SYSMPU->CESR |= SYSMPU_CESR_VLD_MASK;
+	/* Enable MPU */
+	SYSMPU->CESR |= SYSMPU_CESR_VLD_MASK;
 
-		nxp_mpu_enabled = 1U;
-	}
 }
 
 /**
  * @brief disable the MPU
  */
 void arm_core_mpu_disable(void)
 {
-	if (nxp_mpu_enabled == 1) {
-		/* Disable MPU */
-		SYSMPU->CESR &= ~SYSMPU_CESR_VLD_MASK;
-		/* Clear Interrupts */
-		SYSMPU->CESR |=  SYSMPU_CESR_SPERR_MASK;
-
-		nxp_mpu_enabled = 0U;
-	}
+	/* Disable MPU */
+	SYSMPU->CESR &= ~SYSMPU_CESR_VLD_MASK;
+	/* Clear Interrupts */
```

Comment:
As far as I saw in the reference manual, this does not "clear interrupts", but it clears the MPU error status". If agreed, we could re-work the inline comment.

Reply:
Agreed. Could you please update the comment in this patch?

Reply:
Will do.

Reply:
Fixed now.